pkg/cloud/services/networking/service.go

```diff
@@ -44,8 +47,21 @@ func NewService(client *gophercloud.ProviderClient, clientOpts *clientconfig.Cli
 	if err != nil {
 		return nil, fmt.Errorf("failed to create networking service client: %v", err)
 	}
+
+	var projectID string
+	if clientOpts.AuthInfo != nil {
```

Comment:
is it possible clientOpts.AuthInfo is nil? 
if so the projectID will be "" , will it impact follow up features?

Reply:
I think it cannot be nil, but it has very high impact if it is. With an empty project id we would just list and delete orphaned ports depending on the rights of the user.

This means when you have some kind of admin / domain admin credentials in the cloud.yaml it will delete stuff all over the OpenStack.

I think we should fail hard here if we're unable to determine a project id

Reply:
ok, thanks for the info